docs/_static/spec/server.yaml

```
@@ -0,0 +1,1517 @@
+    conversations as well as endpoints to modify them. Additionally,
+    enpoints for training and evaluating models are provided.
+servers:
+  - url: "http://localhost:5005"
+    description: "Local development server"
+
+x-reuse:
+  tracker:
+    &tracker-example
+    conversation_id: default
+    latest_event_time: 1537645578.314389
+    latest_input_channel: rest
+    paused: false
+    slots: {}
+    followup_action:
+    latest_message:
+      text: hello
+      entities: []
+      intent:
+        confidence: 0.57
+        name: greet
+      intent_ranking:
+      - confidence: 0.57
+        name: greet
+      - confidence: 0.33
+        name: goodbye
+    events:
+    - event: action
+      name: action_listen
+      timestamp: 1537645578.274815
+    - event: user
+      input_channel: rest
+      text: hello
+      timestamp: 1537645578.314389
+      parse_data:
+        text: hello
+        entities: []
+        intent:
+          confidence: 0.57
+          name: greet
+        intent_ranking:
+        - confidence: 0.57
+          name: greet
+        - confidence: 0.33
+          name: goodbye
+
+paths:
+  /:
+    get:
+      tags:
+      - Server Information
+      summary: Health endpoint of Rasa Server
+      description: >-
+        This URL can be used as an endpoint to run
+        health checks against. When the server is running
+        this will return 200.
+      responses:
+        200:
+          description: Up and running
+          content:
+            text/plain:
+              schema:
+                type: string
+                description: Welcome text of Rasa Server
+              example: >-
+                Hello from Rasa: 0.12.0
+
+  /version:
+    get:
+      security:
+      - TokenAuth: []
+      - JWT: []
+      tags:
+      - Server Information
+      summary: Version of Rasa
+      description: >-
+        Returns the version of Rasa.
+      responses:
+        200:
+          description: Version of Rasa
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  version:
+                    type: string
+                    description: >-
+                      Core version number
+                  minimum_compatible_version:
+                    type: string
+                    description: >-
+                      Minimum version this Rasa version is
+                      able to load models from
+              example:
+                version: 0.12.0
+                minimum_compatible_version: 0.12.0
+        401:
+          $ref: '#/components/responses/401NotAuthenticated'
+        403:
+          $ref: '#/components/responses/403NotAuthorized'
+
+  /status:
+    get:
+      security:
+      - TokenAuth: []
+      - JWT: []
+      tags:
+      - Server Information
+      summary: Status of the currently loaded Rasa Core & NLU models
```

Given that we remove the projects, there will always only be one model loaded
```suggestion
      summary: Status of the currently loaded Rasa model
```

---

Just for clarification: The server should be able to handle a
1. combined model
2. Rasa NLU model
3. Rasa Core model

Is that correct?

So, depending on what model is currently loaded, some endpoints may not work (e.g. `/conversation` endpoints if nlu model is loaded).

---

Yes that is expected. We should have a standard response code & message we use to signal that.

---

What about 
```
409 Conflict
This response is sent when a request conflicts with the current state of the server.
```

---

I guess this would be the best use for the `I am a teapot` response :D - but yes you are right, 409 sounds good